src/app/myworkflows/my-workflow/my-workflow.component.ts

```diff
@@ -145,6 +145,10 @@ export class MyWorkflowComponent extends MyEntry implements OnInit {
           const sortedSharedWorkflows = this.myworkflowService.sortGroupEntries(sharedWorkflows, this.user.username, 'workflow');
           this.setGroupSharedEntriesObject(sortedSharedWorkflows);
 
+          // If a user navigates directly to an unpublished workflow on their my-workflows page (via bookmark, refresh), the url needs to be
```

Comment:
How does this work if you don't directly navigate to the my-workflows page? Has something else already invoked `workflowService.setWorkflow`?

Reply:
Do you mean if you don't directly navigate to a specific workflow on the my-workflows page (i.e. just go to my-workflows)? If so, the next if statement, `if (!this.workflow)`, will get executed which will eventually set the workflow to your first published entry, or if none are published, just the first workflow.

Reply:
Yes, I meant that. Thanks for the clarification.